src/init/features/dataconnect/index.ts

```diff
@@ -318,23 +313,37 @@ interface serviceAndSchema {
   schema?: Schema;
 }
 
+/**
+ * Picks create new service or an existing service from the list of services.
+ *
+ * Firebase Console can provide `FDC_CONNECTOR` or `FDC_SERVICE` environment variable.
+ * If either is present, chooseExistingService try to match it with any existing service
+ * and short-circuit the prompt.
+ *
+ * `FDC_SERVICE` should have the format `<location>/<serviceId>`.
+ * `FDC_CONNECTOR` should have the same `<location>/<serviceId>/<connectorId>`.
+ * @param existing
+ */
 async function chooseExistingService(
   existing: serviceAndSchema[],
 ): Promise<serviceAndSchema | undefined> {
-  const [serviceLocationFromEnvVar, serviceIdFromEnvVar] = serviceEnvVar().split("/");
-  const serviceFromEnvVar = existing.find((s) => {
-    const serviceName = parseServiceName(s.service.name);
-    return (
-      serviceName.serviceId === serviceIdFromEnvVar &&
-      serviceName.location === serviceLocationFromEnvVar
-    );
-  });
-  if (serviceFromEnvVar) {
-    // FDC_CONNECTOR or FDC_SERVICE env var match an existing service.
-    logBullet(
-      `Picking up the existing service ${clc.bold(serviceLocationFromEnvVar + "/" + serviceIdFromEnvVar)}.`,
-    );
-    return serviceFromEnvVar;
+  const serviceEnvVar = envOverride("FDC_CONNECTOR", "") || envOverride("FDC_SERVICE", "");
+  if (serviceEnvVar) {
+    const [serviceLocationFromEnvVar, serviceIdFromEnvVar] = serviceEnvVar.split("/");
+    const serviceFromEnvVar = existing.find((s) => {
+      const serviceName = parseServiceName(s.service.name);
+      return (
+        serviceName.serviceId === serviceIdFromEnvVar &&
+        serviceName.location === serviceLocationFromEnvVar
+      );
+    });
+    if (serviceFromEnvVar) {
+      logBullet(
+        `Picking up the existing service ${clc.bold(serviceLocationFromEnvVar + "/" + serviceIdFromEnvVar)}.`,
+      );
+      return serviceFromEnvVar;
+    }
+    logWarning(`Unable to pick up an existing service based on FDC_SERVICE=${serviceEnvVar}.`);
```

Comment:
I think we can skip logging the env var if it's simply not set, but otherwise lgtm

Reply:
If it's not set, `serviceEnvVar` is "", so should skip this branch above.